File: 120-triangle/120-triangle.cpp
class Solution {
public:
    int minimumTotal(vector<vector<int>>& triangle) {
        int m = triangle.size(), n = triangle.back().size();
        
        for(int i = 1; i < m; i++) {
            triangle[i-1].push_back(INT_MAX);
            triangle[i][0] += triangle[i-1][0];
            for(int j = 1; j < triangle[i].size(); j++) {
                triangle[i][j] += min(triangle[i-1][j-1], triangle[i-1][j]);
            }
        }
        
        int ans = INT_MAX;
        for(int num: triangle.back()) {
            ans = min(ans, num);
        }
        
        return ans;
    }
};

File: 120-triangle/120-triangle.py
class Solution:
    def minimumTotal(self, tri: List[List[int]]) -> int:
        for i in range(1, len(tri)):
            tri[i][0] += tri[i-1][0]
            for j in range(1, i):
                tri[i][j] += min(tri[i-1][j-1], tri[i-1][j])
            tri[i][i] += tri[i-1][i-1]
        
        return min(tri[-1])